lib/extract/netcdf/kernel_data_netcdf.f90

```diff
@@ -0,0 +1,278 @@
+module psy_data_mod
+
+    implicit none
+
+    ! Contains functionality to write and read variables
+    type, public:: PSyDataType
```

Comment:
Please add doxygen markup to document the various elements of this type.

Reply:
All done, the whole file is properly (I hope) documented now.